### DISCUSSION TITLE
Telling Cockpit to listen on a specific IP address

### DISCUSSION THREAD

**Discussion by yrro** [cockpit-project/cockpit#21193] [Show and tell]
To solve a conflict between cockpit and prometheus, which both want to use port 9090/tcp, I added an additional IP address to my machine, and told prometheus to listen on `192.0.2.10:9090`. However, this still conflicts with `cockpit.socket` which is unable to bind to `0.0.0.0:9090` if prometheus starts first.

I modified `cockpit.socket` to make it bind to the machine's original IP address:

```
[Socket]
FreeBind=true
ListenStream=
ListenStream=192.0.2.5:9090
```

I added `FreeBind=` so that `cockpit.socket` can start before the IP address is available. Originally I tried adding `Wants=network-online.target` but this caused an ordering cycle (`network-online.target` -> `NetworkManager-wait-online.service` -> `basic.target` -> `sockets.target` -> `cockpit.socket` -> `network-online.target`).

I hope someone else finds this useful.

**Comment by yrro** [comment]
Hah, I see the info about `FreeBind=` is already in [the guide](https://cockpit-project.org/guide/latest/listen)! I searched the discussions, but didn't see it in the manual... 🤦

**Comment by martinpitt** [comment]
Thanks @yrro ! Just for other readers, another way is to change cockpit's port to something else. https://cockpit-project.org/guide/latest/listen describes that as well.
